### PR TITLE
skip fields which are empty or only have the default value

### PR DESCRIPTION
**Pull request by arnej27959**
@geirst please review and merge
@bratseth @havardpe FYI

### CODE REVIEW

File: searchsummary/src/vespa/searchsummary/docsummary/docsumwriter.cpp [hunk 1]
```diff
@@ -78,6 +78,9 @@ DynamicDocsumWriter::resolveInputClass(ResolveClassInfo &rci, uint32_t id) const
     }
 }
 
+constexpr uint32_t default_32bits_int = (uint32_t)std::numeric_limits<int32_t>::min();
```
Comment (geirst) [inline searchsummary/src/vespa/searchsummary/docsummary/docsumwriter.cpp]: I suggest we use getUndefined() in searchcommon/src/vespa/searchcommon/common/undefinedvalues.h for this.

**Review by geirst — CHANGES_REQUESTED** [review]
Overall looks good, please have a look at my comment

File: searchsummary/src/vespa/searchsummary/docsummary/docsumwriter.cpp (hunk 1 above)
Comment (arnej27959) [reply searchsummary/src/vespa/searchsummary/docsummary/docsumwriter.cpp]: done.

**Comment by arnej27959** [comment]
@geirst please take another look now.
adding some constexpr made the compiler notice an (existing) overflow in grouping_test, so
@toregge please take a look at that.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**